apis/python/src/tiledb/vector_search/index.py

```diff
@@ -566,6 +541,153 @@ def clear_history(
                 raise ValueError(f"Unsupported index_type: {index_type}")
             group.close()
 
+    def get_dimensions(self):
+        """
+        Returns the dimension of the vectors in the index.
+        """
```

Comment:
Should we do this to match `query_internal()`?
```suggestion
        """
        Abstract method implemented by all Vector Index implementations.
        
        Returns the dimension of the vectors in the index.
        """
```